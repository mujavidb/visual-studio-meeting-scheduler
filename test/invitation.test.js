import React from 'react';
import renderer from 'react-test-renderer';
import Invitation from '../src/components/invitation'

const testData = {
	details: {
		meetingId: "asjdaifhwijd",
		meetingName: "Test",
		finalDate: {
			dateStart: new Date(2011,10,30)
		},
		meetingLocation: "somewhere",
		attendees: [
			{
				id: "92hd98fhueohi",
				status: "unresponsive",
				displayName: "Mr T",
			}
		]
	},
	teamMembers: [
		{
			id: "92hd98fhueohi",
			status: "unresponsive",
			displayName: "Mr T",
			imgUrl: "link"
		}
	],
	ctrl: {
		respondInvitation: () => { }
	}
}

test('Clicking invitation navigates to RespondInvitation', () => {
	const component = renderer.create(
		<Invitation
			key={testData.id}
			details={testData.details}
			ctrl={testData.ctrl}
			teamMembers={testData.teamMembers} />
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	tree.props.onClick();
	expect(tree).toMatchSnapshot();
});