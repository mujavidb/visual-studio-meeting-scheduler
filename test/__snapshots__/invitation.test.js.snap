// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clicking invitation navigates to RespondInvitation 1`] = `
<a
  className="meeting_card_container"
  onClick={[Function]}
  role="button"
>
  <div
    className="meeting_card"
  >
    <h3
      className="meeting_title"
    >
      Test
    </h3>
    <p
      className="meeting_datetime"
    >
      Wednesday 30th November, 12:00am
    </p>
    <p
      className="meeting_location"
    >
      somewhere
    </p>
    <div
      className="attendees"
    >
      <div
        className="attendee_block responsive"
        id="92hd98fhueohi"
        style={
          Object {
            "backgroundImage": "url(undefined)",
            "backgroundSize": "cover",
          }
        }
        title="Mr T has responded"
      />
    </div>
  </div>
</a>
`;

exports[`Clicking invitation navigates to RespondInvitation 2`] = `
<a
  className="meeting_card_container"
  onClick={[Function]}
  role="button"
>
  <div
    className="meeting_card"
  >
    <h3
      className="meeting_title"
    >
      Test
    </h3>
    <p
      className="meeting_datetime"
    >
      Wednesday 30th November, 12:00am
    </p>
    <p
      className="meeting_location"
    >
      somewhere
    </p>
    <div
      className="attendees"
    >
      <div
        className="attendee_block responsive"
        id="92hd98fhueohi"
        style={
          Object {
            "backgroundImage": "url(undefined)",
            "backgroundSize": "cover",
          }
        }
        title="Mr T has responded"
      />
    </div>
  </div>
</a>
`;
